### PR TITLE
Document commit title tags for hotfixes (#3769)

### PR DESCRIPTION
**Pull request by amarjandu**
https://github.com/DataBiosphere/azul/issues/3769

### Author

- [x] PR title references issue
- [x] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
- [x] Title of main commit references issue
- [x] PR is connected to Zenhub issue and description links to issue

### Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

### Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [x] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate       <sub>or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab
- [ ] Moved issues to `prod` or `Merged prod`               <sub>or this PR does not represent a promotion</sub>

### Operator (reindex) 

- [ ] Started reindex in target deployment                  <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures             <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment              <sub>or this PR does not require reindexing</sub>
- [ ] Filed backport PR                                     <sub>or this PR does not represent a hotfix</sub>

### Operator

- [ ] Unassigned PR

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -27,6 +27,12 @@ https://github.com/DataBiosphere/azul/issues/NUMBER
 - [ ] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
 - [ ] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>
 
+## Author (hotfixes)
+
+- [ ] Added `h` tag to commit title                         <sub>or this PR does not include a temporary hotfix</sub>
+- [ ] Added `H` tag to commit title                         <sub>or this PR does not include a permanent hotfix</sub>
```
Comment (amarjandu) [inline .github/pull_request_template.md:33]: For PL: Should author revert a hotfix in the same PR that adds a permanent fix to the issue?

Comment (amarjandu) [reply .github/pull_request_template.md:33]: From PL: "reversion of temp hotfix should proceed the permanent fix for the issue"

**Review by noah-aviel-dove — APPROVED** [review]
Looks good :+1:

**Comment by noah-aviel-dove** [comment]
Checklist is incomplete

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -856,6 +856,11 @@ Commit title tags
   beginning of a commit message. Multiple tags are separated by space. The
   following tags are defined:
 
+  - ``h`` the commit is a temporary hotfix. These commit should be reverted
```
Comment (hannes-ucsc) [inline CONTRIBUTING.rst:859]: Append this to the list, not prepend.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.rst (hunk 2 above)
Comment (amarjandu) [reply CONTRIBUTING.rst:859]: `Tags must appear in a title in the order they are defined above, as in [u r R 1/2]. This ensures that more consequential tags appear earlier.`

https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#36commit-title-tags

Just to confirm the hotfix commit tag should be after `M/N` commit tag?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -27,6 +27,13 @@ https://github.com/DataBiosphere/azul/issues/NUMBER
 - [ ] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
 - [ ] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>
 
+## Author (hotfixes)
+
+- [ ] Added `h` tag to commit title and PR targets `prod`   <sub>or this PR does not include a temporary hotfix</sub>
+- [ ] Added `H` tag to commit title and PR targets `develop` <sub>or this PR does not include a permanent hotfix</sub>
```
Comment (hannes-ucsc) [inline .github/pull_request_template.md:33]: Unfortunately, this breaks the alignment of the sub tags. You'd have to shorten the text or move all the sub tags over.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 3 above)
Comment (amarjandu) [reply .github/pull_request_template.md]: Added 7cabc77ed331cb46c7d61c51ff11aabf23b485b4 to align all the sub tags to https://github.com/DataBiosphere/azul/blob/3f4863032a586cfbdbd296fb101e6237ef638fd3/.github/pull_request_template.md?plain=1#L6 

Added fixup to move the hotfix section to match.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**